storage/util_1.8.go

```diff
@@ -1,32 +0,0 @@
-// +build go1.8
-
-// Copyright 2017 Microsoft Corporation
-//
-//  Licensed under the Apache License, Version 2.0 (the "License");
-//  you may not use this file except in compliance with the License.
-//  You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-//  Unless required by applicable law or agreed to in writing, software
-//  distributed under the License is distributed on an "AS IS" BASIS,
-//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-//  See the License for the specific language governing permissions and
-//  limitations under the License.
-
-package storage
-
-import (
-	"io"
-	"io/ioutil"
-	"net/http"
-)
-
-func setContentLengthFromLimitedReader(req *http.Request, lr *io.LimitedReader) {
-	req.ContentLength = lr.N
-	snapshot := *lr
-	req.GetBody = func() (io.ReadCloser, error) {
```

Comment:
Removing this means we can no longer rely on it for retrying requests and have to fall back to reading the entire contents of lr into a buffer which seems like a regression (the rest of the change looks good).

Reply:
I don't think this implementation works with retries. There is no rewinding when GetBody is called, so the second retry will fail if the reader was consumed (partially or completely) during the 1st attempt. Here's a demonstration:

https://play.golang.org/p/6gllphBhVK

Reply:
Yep you are right (thanks for the sample BTW it was very helpful).  The copies of LimitReader point to the same underlying reader which is really the thing that needs to be the snapshot.  So the current implementation is broken.  :(

Reply:
My pleasure! I guess some more work need to be done on go-autorest project side for retriable requests, I've mentioned a couple of suggestions here: https://github.com/Azure/go-autorest/issues/157#issuecomment-339113595

 How do you propose to proceed?

Reply:
We are getting ready to release an updated storage SDK which fixes this issue (we use io.ReadSeekers to solve the retry problem).